libs/contrib/Data/Elem/Extra.idr

```diff
@@ -0,0 +1,61 @@
+module Data.Elem.Extra
+
+import Data.List
+import Data.List.Elem
+
+%default total
+
+||| Convenience lemma showing that prepending nil to xs is just xs.
+public export
+appendNilLeftNeutral : (xs : List a) -> ([] ++ xs) = xs
```

Comment:
This should be moved to a more sensible location as it has nothing to do with `Elem`.

Reply:
Is it too much to have it go into Data.List next ```appendNilRightNeutral```?
If so, I'll find a spot in contrib/Data.List to put it :)

Reply:
That seems like a sensible place to put it.

Reply:
Btw, you should not need to pattern-match on `xs` here: it should always be
true by computation only.

Reply:
Is there any use of this function? I mean, it's obvious to the compiler and it's not used in newly added lemmata.

Reply:
Ahh whoops, I forgot that I rewrote things to not use it....well looks like I'll remove it again!

Reply:
It can be useful as documentation.